utils/src/main/java/io/atomix/utils/concurrent/OrderedFuture.java

```diff
@@ -45,19 +45,22 @@ public OrderedFuture() {
    * Adds a new ordered future.
    */
   private CompletableFuture<T> orderedFuture() {
-    synchronized (orderedFutures) {
```

Comment:
fields are not volatile and complete, error, result were guarded by this sync, not thread safe anymore?

Reply:
that's my mistake, they should all be volatile